recipes/aerospike-client-c/all/conanfile.py

```diff
@@ -0,0 +1,118 @@
+import os
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration, ConanException
+from conan.tools.files import copy, apply_conandata_patches, export_conandata_patches
+from conan.tools.scm import git
+from conan.tools.layout import basic_layout
+
+required_conan_version = ">=1.57.0"
+
+
+class AerospikeConan(ConanFile):
+    name = "aerospike-client-c"
+    homepage = "https://github.com/aerospike/aerospike-client-c"
+    description = "The Aerospike C client provides a C interface for interacting with the Aerospike Database."
+    topics = ("aerospike", "client", "database")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "event_library": ["libev", "libuv", "libevent", None]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "event_library": None,
+    }
+
+    def validate(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration(
+                "This recipe is not compatible with Windows")
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        self.requires("openssl/[>=1.1 <4]")
+        self.requires("zlib/[>=1.2.11 <2]")
+
+        # in the original code lua is used as a submodule.
+        # when creating a new version, you need to manually check which version of lua is used in the submodule.
+        if self.version == "6.6.0":
+            self.requires(f"lua/5.4.6")
+
+        if self.options.event_library == "libev":
+            self.requires("libev/[>=4.24 <5]")
+        elif self.options.event_library == "libuv":
+            self.requires("libuv/[>=1.15.0 <2]")
+        elif self.options.event_library == "libevent":
+            self.requires("libevent/[>=2.1.8 <3]")
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def layout(self):
+        basic_layout(self, src_folder='src')
+
+    def source(self):
```

Comment:
Please, use source to download a tarball version instead. Using git is not allowed. Please, read https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/folders_and_files.md#the-recipe-folder

Reply:
@ybogo Each submodule should be split as a Conan package in a separated PR.